app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -131,9 +131,13 @@ def alternate_content_sources
         SmartProxy.joins(:smart_proxy_alternate_content_sources).where('katello_smart_proxy_alternate_content_sources.smart_proxy_id' => self.id)
       end
 
-      def registration_host
+      def registration_url
         url = self.setting('Registration', 'registration_url').presence || self.url
-        URI.parse(url).host
+        URI(url)
```

Comment:
I added the bare `registration_url` method here so that Foreman can do things to the `URI` object it returns, directly (e.g. using its `path` and `port`).